Remove DNX branding from footer

JunTaoLuo opened the pull request:
#722 I also did a quick search for other occurrences of the word dnx. Still need to investigate the use of https://github.com/aspnet/Hosting/blob/12fff1833fdcaa36b6d43bbdeb0465a11d9cf365/src/Microsoft.AspNetCore.Server.Testing/Deployers/IISExpressDeployer.cs#L88-L95.

The new error footer looks like:
![image](https://cloud.githubusercontent.com/assets/2030323/14793341/72713e72-0ad3-11e6-9c81-4a8cadacba71.png)


Review: @Tratcher 
Approval: @Eilon @DamianEdwards

davidfowl commented:
The version is wrong but it's the "rightest" thing we can do right now 😄 

:shipit: !

DamianEdwards commented:
Shouldn't be ".NET Framework CoreCLR", that's combining two things that shouldn't be. It's either ".NET Framework version 4.*" or ".NET Core version *.*".

Also, the "x64 Runtime version 4.0.0.0" is confusing. What is that referring to exactly? We don't have DNX anymore so I don't think it's even needed. Move the bitness string into the first string and remove that section completely.

For the OS string, please include the bitness there too, as it may be different to the runtime bitness.

JunTaoLuo commented:
@DamianEdwards Just want to clarify a few things here. 

1. ".NET Framework" and ".NET Core" makes sense but what about Mono? Since we differentiate it in `PlatformAbstractions`, we originally had ".NET Framework Mono" but maybe we can drop the "Mono" part of it. 
2. The runtime version is what we get from `PlatformAbstractions`. There it uses the version of the assembly containing the type `object` as the "Runtime" version. I'm guessing this is what was meant by it's wrong but it's the rightest thing for now.
3. I don't think hosting is the right place to figure out the OS bitness. `PlatformAbstractions` feels like the more logical place to have this logic. There is a work item to update to using CoreFX's `RuntimeInformation` which will have this information but that's for RTM. Unless we are going to pull that back to rc2, I'll look into some other options.

DamianEdwards commented:
Let's just fix the first one for RC2 then. Happy to keep "Mono" where applicable, so ".NET Framework", ".NET Core", and ".NET Framework (Mono)"?

JunTaoLuo commented:
@DamianEdwards I also removed the runtime section as well and moved the bitness string to the clr type section. The updated footer looks like 
![image](https://cloud.githubusercontent.com/assets/2030323/14828045/0aac901c-0b9a-11e6-94f4-ad34ffca1dfe.png)
![image](https://cloud.githubusercontent.com/assets/2030323/14828058/1577bf4e-0b9a-11e6-8093-fb2a6e8fe778.png)
 
Do you like this better?

muratg commented:
I think that's fine, only part that's a little weird is ".NET Framework x64 version 4.0....".  @DamianEdwards thoughts?

davidfowl commented on src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs:
```diff
@@ -234,6 +234,7 @@ private static string GenerateFooterEncoded()
         {
             var environment = PlatformServices.Default.Runtime;
             var runtimeType = HtmlEncodeAndReplaceLineBreaks(environment.RuntimeType);
+            var runtimeDisplayName = runtimeType == "CoreCLR" ? ".NET Core" : runtimeType == "CLR" ? ".NET Framework" : ".NET Framework Mono";
```
Change .NET Framework Mono -> Mono?

JunTaoLuo commented:
:up::date: as per offline discussion with @DamianEdwards 

The updated footer looks like:
![new clr exception](https://cloud.githubusercontent.com/assets/2030323/14862257/65a77c4a-0c65-11e6-85b9-0c7d60a59094.PNG)
![new coreclr exception](https://cloud.githubusercontent.com/assets/2030323/14862258/67d2a940-0c65-11e6-83d1-4c26be136717.PNG)
 Will create a followup issue to convert the full CLR version to something more reasonable like 4.5.1, 4.6...
